numismatic/feeds/luno.py

```diff
@@ -42,13 +42,9 @@ class LunoWebsocketClient(WebsocketClient):
     api_key_secret = attr.ib(default=attr.Factory(
         config_item_getter('LunoFeed', 'api_key_secret')), repr=False)
 
-    async def _connect(self, subscription):
-        websocket_url = f'{self.websocket_url}/{subscription.symbol}'
-        logger.info(f'Connecting to {websocket_url!r} ...')
-        self.websocket = await websockets.connect(websocket_url)
-        if hasattr(self, 'on_connect'):
-            await self.on_connect(self.websocket)
-        return self.websocket
+    def listen(self, symbol, channel=None, websocket_url=None):
+        self.websocket_url = f'{self.websocket_url}/{symbol}'
+        return super().listen(symbol, channel, websocket_url)
```

Comment:
I don't like the fact that `self.websocket_url` is being set in order to pass the info to `_connect()` while the `websocket_url` parameter is unused. I know the `websocket_url` was being ignored in my code already, just saying. The bigger problem in my mind is that `self.websocket_url` is being set by one subscription and then overwritten by another one later on. In my mind if it's not going to be consistent across subscriptions then it shouldn't be state on the instance and should rather be passed as a parameter.